system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -312,67 +312,61 @@ export class _TopSiteList extends React.PureComponent {
         if (index !== this.state.draggedIndex) {
           this.props.dispatch(ac.SendToMain({
             type: at.TOP_SITES_INSERT,
-            data: {site: {url: this.state.draggedSite.url, label: this.state.draggedTitle}, index}
+            data: {site: {url: this.state.draggedSite.url, label: this.state.draggedTitle}, index, draggedFromIndex: this.state.draggedIndex}
           }));
           this.userEvent("DROP", index);
         }
         break;
     }
   }
   _getTopSites() {
-    return this.props.TopSites.rows.slice(0, this.props.TopSitesCount);
+    let topSites = this.props.TopSites.rows.slice(0, this.props.TopSitesCount);
+    if (topSites.length < this.props.TopSitesCount) {
+      topSites[this.props.TopSitesCount - 1] = null;
```

Comment:
I think it would be clearer / simpler to just do
```js
// Make a copy of the sites to truncate or extend to desired length
let topSites = this.props.TopSites.rows.slice();
topSites.length = this.props.TopSitesCount;
return topSites;
```